src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -4,11 +4,25 @@
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
-public class RegistrationServiceImpl implements RegistrationService {
+public class RegistrationServiceImpl {
     private final StorageDao storageDao = new StorageDaoImpl();
 
-    @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserDataException("User cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidUserDataException("login cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException("User with such login already exists");
+        }
+        if (user.getAge() == null || user.getAge() < 18 || user.getAge() < 0) {
+            throw new InvalidUserDataException("User must be at least 18 years old");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserDataException("Password must be at least 6 characters");
+        }
```

Comment:
it's two different checks. create another `if()` that check null password and throw corresponding exception